Command/DumpCommand.php

```diff
@@ -0,0 +1,250 @@
+
+        $backupFilePath = sprintf('%s/%s_backup_%s.sql', rtrim($backupPath, '/'), date('Y-m-d_H-i'), $database);
+        $output->write("<info>»</info> Dumping <info>$database</info> to <info>$backupFilePath</info>");
+
+        $dumpCommand = sprintf('mysqldump -u%s -p%s -h %s -x --compact %s > %s', $username, $password, $host, $database, $backupFilePath);
+
+        $process = new Process($dumpCommand);
+        $process->run();
+
+        $success = $process->isSuccessful();
+        if ($success)
+        {
+            $output->writeln(' ...<info>done</info>');
+        }
+        else
+        {
+            $fileSystem->remove($backupFilePath);
+            $output->writeln(' ...<error>failed</error>');
+        }
+
+        return $success;
+    }
+
+
+    /**
+     * Returns all databases that will be backed up by searching the configuration:
+     *  1) Searches the config.yml for the key 'becklyn_mysql_dump:databases'
+     *  2) Searches the config.yml for the connections set up under 'doctrine:dbal:connections'
+     *
+     * @return array
+     */
+    protected function getDatabases ()
+    {
+        $databases = $this->getConfiguredDatabases();
+
+        if (count($databases) === 0)
+        {
+            $databases = $this->getAppDatabases();
+        }
+
+        return $databases;
+    }
+
+
+    /**
+     * Returns a list of all databases that are configured under becklyn_mysql_dump_databases:
+     *
+     * @return array
+     */
+    protected function getConfiguredDatabases ()
+    {
+        $databases = $this->getContainer()->get('becklyn_mysql_dump.configuration')->getConfig('databases');
+
+        return $this->getConnectionDataByDatabaseNames($databases);
+    }
+
+
+    /**
+     * Returns all databases that are associated with this app.
+     * See {@link http://symfony.com/doc/current/cookbook/doctrine/multiple_entity_managers.html}
+     *
+     * @return array
+     */
+    protected function getAppDatabases ()
+    {
+        $databaseNames = [];
+        $doctrine  = $this->getContainer()->get('doctrine');
+
+        /** @var \Doctrine\DBAL\Connection $connection */
+        foreach ($doctrine->getConnections() as $connection)
+        {
+            $databaseNames[] = $connection->getDatabase();
+        }
+
+        return $this->getConnectionDataByDatabaseNames($databaseNames);
+    }
+
+
+    /**
+     * Looks up the connection data for the given database names
+     *
+     * @param array $databaseNames
+     *
+     * @return array
+     */
+    protected function getConnectionDataByDatabaseNames (array $databaseNames)
+    {
+        $connectionData = [];
+        $doctrine = $this->getContainer()->get('doctrine');
+
+        /** @var \Doctrine\DBAL\Connection $connection */
+        foreach ($doctrine->getConnections() as $connection)
+        {
+            if (in_array($connection->getDatabase(), $databaseNames))
+            {
+                $connectionData[$connection->getDatabase()] = [
```

Comment:
Isn't there a special class for it? You should create one and use it, instead of just simple arrays.